holoviews/element/raster.py

```diff
@@ -616,163 +632,63 @@ class QuadMesh(Raster):
-
-
-    @property
-    def _zdata(self):
-        return self.data[2]
-
+    _binned = True
 
-    def _validate_data(self, data):
-        x, y, z = data
-        if not z.ndim == 2:
-            raise ValueError("Z-values must be 2D array")
-
-        ys, xs = z.shape
-        shape_errors = []
-        if x.ndim == 1 and xs+1 != len(x):
-            shape_errors.append('x')
-        if x.ndim == 1 and ys+1 != len(y):
-            shape_errors.append('y')
-        if shape_errors:
-            raise ValueError("%s-edges must match shape of z-array." %
-                             '/'.join(shape_errors))
-        return data
-
-
-    def __getitem__(self, slices):
-        if slices in self.dimensions(): return self.dimension_values(slices)
-        slices = util.process_ellipses(self,slices)
-        if not self._grid:
-            raise KeyError("Indexing of non-grid based QuadMesh"
-                           "currently not supported")
-        slices = util.wrap_tuple(slices)
-        if len(slices) == 1:
-            slices = slices+(slice(None),)
-        if len(slices) > (2 + self.depth):
-            raise KeyError("Can only slice %d dimensions" % (2 + self.depth))
-        elif len(slices) == 3 and slices[-1] not in [self.vdims[0].name, slice(None)]:
-            raise KeyError("%r is the only selectable value dimension" % self.vdims[0].name)
-        slices = slices[:2]
-        if not isinstance(slices, tuple): slices = (slices, slice(None))
-        slc_types = [isinstance(sl, slice) for sl in slices]
-        if not any(slc_types):
-            indices = []
-            for idx, data in zip(slices, self.data[:self.ndims]):
-                dig = np.digitize([idx], data)
-                indices.append(dig-1 if dig else dig)
-            return self.data[2][tuple(indices[::-1])][0]
-        else:
-            sliced_data, indices = [], []
-            for slc, data in zip(slices, self.data[:self.ndims]):
-                if isinstance(slc, slice):
-                    low, high = slc.start, slc.stop
-                    lidx = (None if low is None else
-                            max((np.digitize([low], data)-1, 0))[0])
-                    hidx = (None if high is None else
-                            np.digitize([high], data)[0])
-                    sliced_data.append(data[lidx:hidx])
-                    indices.append(slice(lidx, (hidx if hidx is None else hidx-1)))
-                else:
-                    index = (np.digitize([slc], data)-1)[0]
-                    sliced_data.append(data[index:index+2])
-                    indices.append(index)
-            z = np.atleast_2d(self.data[2][tuple(indices[::-1])])
-            if not all(slc_types) and not slc_types[0]:
-                z = z.T
-            return self.clone(tuple(sliced_data+[z]))
+    def __setstate__(self, state):
+        """
+        Ensures old-style QuadMesh types without an interface can be unpickled.
 
+        Note: Deprecate as part of 2.0
```

Comment:
Is this being tested via the current notebook tests?

Reply:
It was, but once tests are rebuilt just once it won't be.